src/Common/ZooKeeper/ZooKeeperCommon.cpp

```diff
@@ -898,4 +898,19 @@ ZooKeeperRequestFactory::ZooKeeperRequestFactory()
     registerZooKeeperRequest<OpNum::FilteredList, ZooKeeperFilteredListRequest>(*this);
 }
 
+PathMatchResult matchPath(const std::string_view path, const std::string_view match_to)
+{
+    using enum PathMatchResult;
+
+    auto [first_it, second_it] = std::mismatch(path.begin(), path.end(), match_to.begin(), match_to.end());
+
+    if (second_it != match_to.end())
+        return NOT_MATCH;
+
+    if (first_it == path.end())
+        return EXACT;
+
+    return *first_it == '/' ? IS_CHILD : NOT_MATCH;
+}
```

Comment:
this function seems to be not tolerable to trailing slash on match_to argument - consider:
```
matchPath("/path/file", "/path/");
```
will return `NOT_MATCH`, while it's `IS_CHILD`

Reply:
nice catch, I handled those cases + added a unit test for some combination of paths.